packages/ripple-binary-codec/test/definitions.test.js

```diff
@@ -0,0 +1,103 @@
+const { encode, decode, XrplDefinitions } = require('../src')
+const normalDefinitionsJson = require('../src/enums/definitions.json')
+const { UInt32 } = require('../dist/types/uint-32')
+
+const txJson = {
+  Account: 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
+  Amount: '1000',
+  Destination: 'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh',
+  Fee: '10',
+  Flags: 0,
+  Sequence: 1,
+  TransactionType: 'Payment',
+}
+
+describe('encode and decode using new types as a parameter', function () {
+  test('can encode and decode a new TransactionType', function () {
+    const tx = Object.assign({}, txJson, {
+      TransactionType: 'NewTestTransaction',
+    })
```

Comment:
nit: this would be a little cleaner like this:
```suggestion
    const tx = {...txJson, TransactionType: 'NewTestTransaction'}
```